go/vt/vtgate/planbuilder/join.go

```diff
@@ -132,17 +120,16 @@ func (jb *join) Symtab() *symtab {
 	return jb.symtab.Resolve()
 }
 
-// MaxOrder satisfies the builder interface.
-func (jb *join) MaxOrder() int {
-	return jb.rightMaxOrder
+// Order satisfies the builder interface.
+func (jb *join) Order() int {
+	return jb.order
 }
 
-// SetOrder satisfies the builder interface.
-func (jb *join) SetOrder(order int) {
-	jb.Left.SetOrder(order)
-	jb.leftMaxOrder = jb.Left.MaxOrder()
-	jb.Right.SetOrder(jb.leftMaxOrder)
-	jb.rightMaxOrder = jb.Right.MaxOrder()
+// Reorder satisfies the builder interface.
+func (jb *join) Reorder(order int) {
+	jb.Left.Reorder(order)
+	jb.leftOrder = jb.Left.Order()
+	jb.Right.Reorder(jb.leftOrder)
```

Comment:
It's not obvious to me why we don't need to do something like:

```
jb.order = rb.Right.Order()
```

Are they always the same?